proxy/pkg/middleware/account_resolver.go

```diff
@@ -56,22 +61,15 @@ func (m accountResolver) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	if u == nil && claims != nil {
-		var claim, value string
-		switch {
-		case claims.PreferredUsername != "":
-			claim, value = "username", claims.PreferredUsername
-		case claims.Email != "":
-			claim, value = "mail", claims.Email
-		case claims.OcisID != "":
-			//claim, value = "id", claims.OcisID
-		default:
-			// TODO allow lookup by custom claim, eg an id ... or sub
-			m.logger.Error().Msg("Could not lookup account, no mail or preferred_username claim set")
-			w.WriteHeader(http.StatusInternalServerError)
-		}
 
 		var err error
-		u, err = m.userProvider.GetUserByClaims(req.Context(), claim, value, true)
+		if value, ok := claims[m.userOIDCClaim].(string); ok && value != "" {
+			u, err = m.userProvider.GetUserByClaims(req.Context(), m.userCS3Claim, value, true)
+		} else {
+			m.logger.Error().Str("claim", m.userOIDCClaim).Interface("claims", claims).Msg("claim not set or empty")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 
 		if m.autoProvisionAccounts && err == backend.ErrAccountNotFound {
 			m.logger.Debug().Interface("claims", claims).Interface("user", u).Msgf("User by claim not found... autoprovisioning.")
```

Comment:
- [ ] the user can only be nil at this point
- [ ] log configured oidc and cs3 claim
- [ ] no msgf

Reply:
- [ ] do we still need autoprovisioning? it was part of the old migration strategy AFAIR

Reply:
leaving it in for now